src/common/utils/stringer.hpp

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef STRINGER_HPP_
+#define STRINGER_HPP_
+
+#include <cstddef>
+
+/**
+ * Interface used to convert derived type to string.
+ */
+class Stringer {
+public:
+    /**
+     * Returns string representation of derived class.
+     *
+     * @return string.
+     */
+    virtual const char* ToString() const = 0;
+};
+
+/**
+ * Template used to convert enum to strings.
+ */
+template <class T>
+class EnumStringer : public Stringer {
+public:
+    /**
+     * Construct a new EnumStringer object with default type.
+     *
+     * @param value curent enum value.
+     */
+    EnumStringer()
+        : mValue(static_cast<T>(0)) {};
+
+    /**
+     * Construct a new EnumStringer object with specified type.
+     *
+     * @param value curent enum value.
+     */
+    EnumStringer(T value)
+        : mValue(value) {};
+
+    /**
+     * Returns current enum value.
+     *
+     * @return value.
+     */
+    T GetValue() const { return mValue; };
+
+    /**
+     * Compares if EnumStringer equals to another EnumStringer.
+     *
+     * @param stringer EnumStringer to compare with.
+     * @return bool result.
+     */
+    bool operator==(const EnumStringer<T>& stringer) { return GetValue() == stringer.GetValue(); };
+
+    /**
+     * Compares if EnumStringer doesn't equal to another EnumStringer.
+     *
+     * @param stringer EnumStringer to compare with.
+     * @return bool result.
+     */
+    bool operator!=(const EnumStringer<T>& stringer) { return GetValue() != stringer.GetValue(); };
+
+    /**
+     * Compares if EnumStringer equals to specified EnumStringer type.
+     *
+     * @param stringer EnumStringer to compare with.
+     * @return bool result.
+     */
+    bool operator==(T type) { return GetValue() == type; };
+
+    /**
+     * Compares if EnumStringer doesn't equal to specified EnumStringer type.
+     *
+     * @param stringer EnumStringer to compare with.
+     * @return bool result.
+     */
+    bool operator!=(T type) { return GetValue() != type; };
+
+    /**
+     * Compares if specified EnumStringer type equals to EnumStringer.
+     *
+     * @param type specified EnumStringer type.
+     * @param stringer EnumStringer to compare.
+     * @return bool result.
+     */
+    friend bool operator==(T type, const EnumStringer<T>& stringer) { return stringer.GetValue() == type; };
+
+    /**
+     * Compares if specified EnumStringer type doesn't equal to EnumStringer.
+     *
+     * @param type specified EnumStringer type.
+     * @param stringer EnumStringer to compare.
+     * @return bool result.
+     */
+    friend bool operator!=(T type, const EnumStringer<T>& stringer) { return stringer.GetValue() != type; };
+
+    /**
+     * Converts enum value to string.
+     *
+     * @return string.
+     */
+    const char* ToString() const override
```

Comment:
Could we consider returning a Pair consisting of an error and a string, which would allow for the possibility of checking conversions?

Reply:
We need to check if the conversion to a string was successful or not, because if it fails, the log output will show undefined. This can be confusing and it's not clear what it means.

Reply:
This is just a suggestion, so it's entirely up to you.

Reply:
Still do not clearly get the use case. `ToString()` method is used in the log for example like this:
```c++
LOG_DBG() << SomeStringerInstance();
```
Why do we need to check if conversion is successful or not in this case, and what we shall do if the conversion is not successful?